extensions/Khronos/KHR_texture_transform/README.md

```diff
@@ -0,0 +1,102 @@
+   -sin(Rotation), cos(Rotation), 0,
+                0,             0, 1
+);
+mat3 scale = mat3(Scale.x,0,0, 0,Scale.y,0, 0,0,1);
+
+mat3 matrix = translation * rotation * scale;
+vec2 uvTransformed = ( matrix * vec3(Uv.xy, 1) ).xy;
+```
+
+This is equivalent to Unity's `Material#SetTextureOffset` and `Material#SetTextureScale`, or Three.js's `Texture#offset` and `Texture#repeat`. UV rotation is not widely supported as of today, but is included here for forward compatibility.
+
+## glTF Schema Updates
+
+The `KHR_texture_transform` extension may be defined on `textureInfo` structures. It may contain the following properties:
+
+| Name       | Type       | Default      | Description
+|------------|------------|--------------|---------------------------------
+| `offset`   | `array[2]` | `[0.0, 0.0]` | The offset of the UV coordinate origin as a factor of the texture dimensions.
+| `rotation` | `number`   | `0.0`        | Rotate the UVs by this many radians counter-clockwise around the origin. This is equivalent to a similar rotation of the image clockwise.
+| `scale`    | `array[2]` | `[1.0, 1.0]` | The scale factor applied to the components of the UV coordinates.
+| `texCoord` | `integer`  |              | Overrides the textureInfo texCoord value if supplied, and if this extension is supported.
+
+Though this extension's values are unbounded, they will only produce sane results if the texture sampler's `wrap` mode is `REPEAT`, or if the result of the final UV transformation is within the range [0, 1] (i.e. negative scale settings and correspondingly positive offsets).
+
+> **Implementation Note**: For maximum compatibility, it is recommended that exporters generate UV coordinate sets both with and without transforms applied, use the post-transform set in the texture `texCoord` field, then the pre-transform set with this extension. This way, if the extension is not supported by the consuming engine, the model still renders correctly. Including both will increase the size of the model, so if including the fallback UV set is too burdensome, either add this extension to `extensionsRequired` or use the same texCoord value in both places.
+
+### JSON Schema
+
+[KHR_texture_transform.textureInfo.schema.json](schema/KHR_texture_transform.textureInfo.schema.json)
+
+### Example JSON
+
+This example utilizes only the lower left quadrant of the source image, rotated clockwise 90&deg;.
+
+```json
+{
+  "materials": [{
+    "emissionTexture": {
+      "source": 0,
+      "extensions": {
+        "KHR_texture_transform": {
+          "offset": [0, 1],
+          "rotation": 1.57079632679,
+          "scale": [0.5, 0.5]
+        }
+      }
+    }
+  }]
+}
+```
+
+This example inverts the T axis, effectively defining a bottom-left origin.
+
+```json
+{
+  "materials": [{
+    "emissionTexture": {
+      "source": 0,
+      "extensions": {
+        "KHR_texture_transform": {
+          "offset": [0, 1],
+          "scale": [1, -1]
```

Comment:
Are we at all concerned that the `normalTextureInfo` structure already contains a `scale` value? I suppose there could be confusion that the scale value in the texture_transform is an override...
Maybe not but I thought I'd mention it.

Reply:
I think it might be confusing, but I don't have a better suggestion. For reference, here is what it might look like:

```json
      "normalTexture": {
        "index": 2,
        "scale": 1.5,
        "extensions": {
          "KHR_texture_transform": {
            "offset": [0, 1],
            "rotation": 1.57079632679,
            "scale": [0.5, 0.5]
          }
        }
      },
```

Reply:
It feels OK nested under the `*_texture_transform` namespace to me. Conceptually it then becomes `transformScale` or something like that in code.